### ISSUE TITLE
ApolloGateway doesn't recognize federation-jvm hosted graphql

### ISSUE THREAD

**Issue by yuhanz** [apollographql/federation-jvm#21]
` _query { _service { sdl } }` on federation-jvm gives

```
{
  "errors": [],
  "data": {
    "_service": {
      "sdl": "schema {\n  query: Query\n}\n\ntype Query {\n  me: User\n}\n\ntype User @key(fields : \"id\") {\n  id: ID!\n  name: String\n  username: String\n}\n"
    }
  },
  "extensions": null,
  "dataPresent": true
}
```

while ApolloGateway is expecting a different format with the same schema.grapql file:
```
{
  "data": {
    "_service": {
      "sdl": "extend type Query {\n  me: User\n}\n\ntype User @key(fields: \"id\") {\n  id: ID!\n  name: String\n  username: String\n}\n"
    }
  }
}
```
https://github.com/apollographql/federation-demo/blob/master/services/accounts/index.js#L4

ApolloGateway doesn't like the empty Array in `.errors`. ApolloGateway considers this as not having a schema, and it also complains when having `schema {\n  query: Query\n}\n\n` in sdl.

Once the sdl string is patched and the empty array `.errors` is removed, the Graphql schema defined in federation-jvm starts to appears on ApolloGateway.

snippet to patch the sql string:
```
final String modified = sdl.replace("schema {\n  query: Query\n}\n\n", "");
_service.put("sdl", modified);
```

**Comment by yuhanz** [comment]
The first part is a non-issue in the spring-example. (`.errors:[]` was introduced because we serialized `ExecutionResult` directly.)

So, the only problem is with the `schema {\n  query: Query\n}` in the sdl

**Comment by kenshih** [comment]
I believe I ran across the same problem. 
This is what it looks like on [federation-demo](https://github.com/apollographql/federation-demo) (in this case for my sample service called "productExtension"):
```
[nodemon] watching: *.*
[nodemon] starting `node gateway.js`
Encountered error when loading productExtension at http://localhost:9000/graphql: Syntax Error: Expected Name, found }
🚀 Server ready at http://localhost:4000/
```

**Comment by glasser** [comment]
What version of federation-jvm are you using? 0.3.2 is out now. Can you share the exact version of `@apollo/gateway` as well, and ideally the source to your server? We are successfully running federation-jvm in production behind gateway.

**Comment by kenshih** [comment]
Sure! Thanks, David!

I tried it with federation-jvm 0.3.2.
and used [federation-demo](https://github.com/apollographql/federation-demo) which has not changed in awhile, which i now see is using @apollo/gateway:latest, which resolves to 0.6.6

Perhaps it's a problem with `federation-demo`s dependency, since i see gateway is now on 0.10.7. I haven't tried this against a new example.

N.B. I AM able to run today's latest `master` spring-example, if i make these changes to `inventory.graphql`:
```
# OLD inventory.graphql
type Product @key(fields: "upc") @extends {
    upc: String! @external
    inStock: Boolean
    quantity: Int
}
 
# NEW inventory.graphql
# comment out "inStock" because this conflicts with federation-demo
# the "Query @extends" is a work-around from https://github.com/apollographql/federation-jvm/issues/23 
type Product @key(fields: "upc") @extends {
    upc: String! @external
    #inStock: Boolean
    quantity: Int
}
 
type Query @extends {
    trivial: Product
}
```

**Comment by sachindshinde** [comment]
Thanks for the version information! We'll look more into this, and get back to you by March 9th.

**Comment by omarsallah00** [comment]
Hello @kenshih  i tried your solution but i got this error

Error: Unknown type: "Product".
    at ASTDefinitionBuilder._resolveType (c:\NWLS\workspace-ASK\federation-demo-master\node_modules\graphql\utilities\extendSchema.js:113:13)
    
I'm using federation-jvm 0.3.2

And 

"@apollo/federation": "^0.10.2",
    "@apollo/gateway": "^0.10.7",
    "apollo-server": "^2.8.2",
     "apollo-server-lambda": "^2.8.2",
     "graphql": "^14.5.0"

Any thoughts?

**Comment by sachindshinde** [comment]
@kenshih @areidt84 @omarsallah00 I believe your issues are similar to #23 , which ended up being a bug in `graphql-java`'s `SchemaPrinter`. We've patched their printer in `federation-jvm` v0.3.4 and above, let me know if you're still experiencing issues and I can reopen this issue.
